airflow/models.py

```diff
@@ -869,7 +877,17 @@ def record_failure(self, error, test_mode=False):
         except Exception as e2:
             logging.error(
                 'Failed to send email to: ' + str(task.email))
-            logging.error(str(e2))
+            logging.exception(e2)
+
+        # Handling callbacks pessimistically
+        try:
```

Comment:
it might be nice to also always email when the email is set.

Would it be possible to have callbacks be a list and always seed the list with the email callback?